public/jsx/components/card-form.jsx

```diff
@@ -135,22 +135,28 @@ module.exports = React.createClass({
           console.log('Successfully subscribed + completed payment');
           router.transitionTo('complete');
         }).fail(function($xhr) {
-          var errors = this.processApiErrors($xhr.responseJSON);
-          this.setState({
-            isSubmitting: false,
-            errors: errors,
-          });
+          this.processApiErrors($xhr.responseJSON);
         });
       }
     });
   },
 
+  errorKeyToFieldMap: {
+    '__all__': {
+      field: 'card',
+      error: 'declined',
+    },
+    'fraud': {
+      field: 'card',
+      error: 'declined',
+    },
+    'cvv': {
+      field: 'cvv',
+      error: 'invalid',
+    },
+  },
+
   processApiErrors: function(errors) {
-    errors = errors || {};
-    var newErrors = {
-      fields: {},
-      generic: [],
-    };
     var that = this;
     if (errors.error_response && errors.error_response.braintree) {
```

Comment:
Those would be the form-level unexpected errors I think as above. I'll work that in when I get to it.